### PR TITLE
3 game management part

### LINKED ISSUE #3: Add game management part
Add every components which permit to control the game flow and every actions.

### PR DESCRIPTION
**Pull request by nicolasGibaud7**

### CODE REVIEW

File: Classic_RPG/src/game/Classic_RPG.java [hunk 1]
```diff
@@ -1,26 +1,108 @@
...
-		m1.basicHit(player1);
-		m1.basicHit(player1);
-		m1.basicHit(player1);
+		System.out.println("How will this courageous aventurer named ?");
+		String name = scan.nextLine();
+		System.out.println("Oh ! " + name + " what a fabulous name !");
 		
-		System.out.println("\nPlayer HP :" + player1.hp +  "\nMonster HP :" + m1.hp);
+		do {
+			System.out.println("Player generation :\nPlease choose a class :\n\t1)Warrior\n\t2)Wizard\n\t3)Ninja");
```
Comment (nicolasGibaud7) [inline Classic_RPG/src/game/Classic_RPG.java]: We could change \t by 2 spaces

File: Classic_RPG/src/game/Classic_RPG.java [hunk 2]
```diff
@@ -1,26 +1,136 @@
...
+			System.out.println("New level !\nLevel n°" + currentLevel.getNumber() + " : " + currentLevel.getName());
+			for (Wave currentWave : currentLevel.getWaves()) {
+				System.out.println("New wave ! Wave n°" + currentWave.getNumber());
+				int turnIndex = 1;
+				boolean playerAlive = true;
 
+				while(!currentWave.getMonsters().isEmpty() && playerAlive) {
+					System.out.println("\nTurn n°" + turnIndex);
+					currentTurn = new Turn(turnIndex, currentWave, player);
+					currentTurn.run_turn();
```
Comment (LavergneC) [inline Classic_RPG/src/game/Classic_RPG.java:27]: add playerAlive = ...

File: Classic_RPG/src/game/Entity.java [hunk 3]
```diff
@@ -62,6 +62,12 @@ protected void attack(Entity target, int dmgs)
 	{
 		target.receiveAttack(dmgs);
 	}
+	
+	void rest()
+	{
+		stamina += (int)Math.ceil( stamina/10 ); // TODO could be changed or implement in daughter class
+		hp += (int)Math.ceil( hp/20 );
+	}
```
Comment (nicolasGibaud7) [inline Classic_RPG/src/game/Entity.java]: It is not max stamina instead of stamina.

File: Classic_RPG/src/game/Entity.java [hunk 4]
```diff
@@ -50,19 +53,29 @@ protected int getMana() {
...
-	
+
 	protected void attack(Entity target, int dmgs)
 	{
 		target.receiveAttack(dmgs);
 	}
 
+	void rest()
+	{
+		if(stamina + (int)Math.ceil(characteristics.getMax_stamina()/10) > characteristics.getMax_stamina())
```
Comment (LavergneC) [inline Classic_RPG/src/game/Entity.java]: todo, changeStamina

File: Classic_RPG/src/game/Turn.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		}
+		
+		/* Monsters turn */
+		Iterator<Monster> it = this.actual_wave.getMonsters().iterator();
+		
+		while(it.hasNext()) {
+			delaySec(2);
+			Monster monster = it.next();
+			if(monster.getStamina() < 50) { /* not enough stamina to attack */
+				System.out.println(monster.getName() + " take some rest.");
```
Comment (LavergneC) [inline Classic_RPG/src/game/Turn.java]: put print into parent rest method

File: Classic_RPG/src/game/Turn.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+				
+				//TODO add new attacks when created
+			}
+			/* Remove monster from the wave if we get hp != 0 */
+			if (actual_wave.getMonsters().get(index_monstre).getHp() <= 0)
+				actual_wave.removeMonster(index_monstre);
+			break;
+			
+		case REST:
+			System.out.println("Player " + player.getName() + " take some rest.");
```
Comment (LavergneC) [inline Classic_RPG/src/game/Turn.java]: cf next comment

**Review by LavergneC — COMMENTED** [review]
wait for corrections for merge

File: Classic_RPG/src/game/Turn.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+		boolean is_valid = true;
+		Action action = null;
+		
+		do {
+			System.out.println(player.toString());
+			for(Monster monster : actual_wave.getMonsters()) {
+				System.out.println(monster.toString());
+			}
+			
+			System.out.println("\nWhich action ?\n  1) Attack\n  2) Rest\n  3) Defense\n");
```
Comment (nicolasGibaud7) [inline Classic_RPG/src/game/Turn.java]: We should create a method in Wave class to do that and call it hear.

**Review by nicolasGibaud7 — COMMENTED** [review]
Still some tiny changes to do

**Pull request merged**